### DISCUSSION TITLE
Docker & Redis: [ioredis] Unhandled error event: undefined

### DISCUSSION THREAD

**Discussion by HugoRCD** [nitrojs/nitro#2862] [Help Wanted]
### Environment

Docker
Bun preset

### Reproduction

Link from docker-compose
https://github.com/HugoRCD/shelve/blob/main/docker/docker-compose.local.yml

Link to repohttps://github.com/HugoRCD/shelve

### Describe the bug

I have this error `[ioredis] Unhandled error event: undefined` which only happens to me when using Docker because locally it works perfectly (obviously 🥲) and also perfectly on Vercel

**Comment by pi0** [comment]
Since `shelve-network` is on bridge, is there any chance that it might be port conflicting with other services (or redis on host) perhaps?

**Reply by HugoRCD** [reply]
I attempted to specify the ports to prevent conflicts with the host, but unfortunately, it didn't work. As for other services, there are no additional ones beyond those listed in the docker-compose.

**Reply by HugoRCD** [reply]
Ok so I got it to work but the solution seems odd I just added an 'or' condition if it doesn't see the environment variable but it looks more like a workaround than a real solution. @pi0 

```typescript
nitro: {
    preset: process.env.NITRO_PRESET || 'bun',
    storage: {
      cache: {
        driver: 'redis',
        url: process.env.NUXT_PRIVATE_REDIS_URL || 'redis://shelve_redis:6379',
      },
    }
  },
```

**Reply by HugoRCD** [reply]
This is strange because it's the only variable that doesn't function in the app, even though she's set in the compose. Here is her login in the container:

![CleanShot 2024-11-10 at 20 25 12](https://github.com/user-attachments/assets/893f0605-694a-495c-a2d1-46629b2b1031)
